sway-fmt-v2/src/config/expr.rs

```diff
@@ -0,0 +1,115 @@
+    pub expr_brace_style: ExprBraceStyle,
+    /// Add trailing semicolon after break, continue and return.
+    pub trailing_semicolon: bool,
+    /// Leave a space before the colon.
+    pub space_before_colon: bool,
+    /// Leave a space after the colon.
+    pub space_after_colon: bool,
+
+    /////OPERATORS/////
+    /// Determines if `+` or `=` are wrapped in spaces in the punctuation of types.
+    pub type_combinator_layout: TypeCombinatorLayout,
+    /// Put spaces around the `..` and `..=` range operators.
+    pub spaces_around_ranges: bool,
+
+    /////MATCH EXPR/////
+    /// Put a trailing comma after a block based match arm (non-block arms are not affected).
+    pub match_block_trailing_comma: bool,
+    /// Determines whether leading pipes are emitted on match arms.
+    pub match_arm_leading_pipe: MatchArmLeadingPipe,
+
+    /////FUNCTIONS/////
+    /// Force multiline closure bodies and match arms to be wrapped in a block.
+    pub force_multiline_blocks: bool,
+    /// Control the layout of arguments in a function.
+    pub fn_args_layout: ItemsLayout,
+    /// Put single-expression functions on a single line.
+    pub fn_single_line: bool,
+}
+
+impl Default for Expressions {
+    fn default() -> Self {
+        Self {
+            expr_brace_style: ExprBraceStyle::AlwaysSameLine,
+            trailing_semicolon: true,
+            space_before_colon: false,
+            space_after_colon: false,
+            type_combinator_layout: TypeCombinatorLayout::Wide,
+            spaces_around_ranges: false,
+            match_block_trailing_comma: false,
+            match_arm_leading_pipe: MatchArmLeadingPipe::Never,
+            force_multiline_blocks: false,
+            fn_args_layout: ItemsLayout::Tall,
+            fn_single_line: false,
+        }
+    }
+}
+
+impl Expressions {
+    pub fn from_opts(opts: &ExpressionsOptions) -> Self {
+        Self {
+            expr_brace_style: opts
+                .expr_brace_style
+                .unwrap_or(ExprBraceStyle::AlwaysSameLine),
+            trailing_semicolon: opts.trailing_semicolon.unwrap_or(true),
+            space_before_colon: opts.space_before_colon.unwrap_or(false),
+            space_after_colon: opts.space_after_colon.unwrap_or(false),
+            type_combinator_layout: opts
+                .type_combinator_layout
+                .unwrap_or(TypeCombinatorLayout::Wide),
+            spaces_around_ranges: opts.spaces_around_ranges.unwrap_or(false),
+            match_block_trailing_comma: opts.match_block_trailing_comma.unwrap_or(false),
+            match_arm_leading_pipe: opts
+                .match_arm_leading_pipe
+                .unwrap_or(MatchArmLeadingPipe::Never),
+            force_multiline_blocks: opts.force_multiline_blocks.unwrap_or(false),
+            fn_args_layout: opts.fn_args_layout.unwrap_or(ItemsLayout::Tall),
+            fn_single_line: opts.fn_single_line.unwrap_or(false),
+        }
```

Comment:
Currently, the default values for the `Expressions` fields are manually specified in two places:

1. In the implementation of `Default` for `Expressions` and
2. In the implementation of `Expressions::from_opts`.

This makes it easy to make mistakes when updating default values in the future. E.g. a value may only get updated in one place, but not the other. To avoid this, there should be a single source of truth for each default value.

One approach would be to specify each default value as a constant, and use the constant in both implementations.

Another way of achieving this might be to leave the `Expressions::default` implementation and use it within the `Expressions::from_opts`, so `Expressions::from_opts` would look something like:

```rust
impl Expressions {
    pub fn from_opts(opts: &ExpressionsOptions) -> Self {
        let default = Self::default();
        Self {
            expr_brace_style: opts
                .expr_brace_style
                .unwrap_or(default.expr_brace_style),
            trailing_semicolon: opts.trailing_semicolon.unwrap_or(default.trailing_semicolon),
            // etc
```

Whatever solution we use, we should apply it to the other sets of configuration options too.